### PR TITLE
adding GetUsersByEmail feature

### PR DESCRIPTION
**Pull request by vr001**
Hey there! So you want to contribute to a Firebase SDK? 
Before you file this pull request, please read these guidelines:

### Discussion

With reference to issue https://github.com/firebase/firebase-admin-go/issues/138. 

### Testing

Unit test and integration tests passed, but haven't added test for the new function GetUsersByEmail.

### CODE REVIEW

**Comment by CDDelta** [comment]
Thanks for doing this but I think there are some ways to optimize this function.

File: auth/user_mgt.go [hunk 1]
```diff
@@ -256,6 +256,25 @@ func (c *Client) GetUserByEmail(ctx context.Context, email string) (*UserRecord,
...
 
+// GetUsersByEmail gets the users data corresponding to the specified emails.
+func (c *Client) GetUsersByEmail(ctx context.Context, emails []string) ([]*UserRecord, error) {
+	var userRecords []*UserRecord
+	for _, email := range emails {
+		if err := validateEmail(email); err != nil {
+			return nil, err
+		}
+		request := &identitytoolkit.IdentitytoolkitRelyingpartyGetAccountInfoRequest{
+			Email: []string{email},
```
Comment (CDDelta) [inline auth/user_mgt.go]: You can pass the entire slice of emails to the request and get all the account info back at once.

Comment (vr001) [reply auth/user_mgt.go]: I am using the existing function https://github.com/firebase/firebase-admin-go/blob/master/auth/user_mgt.go#L642 whose return type is *UserRecord. 
As per what you suggested, probably I would need to create a separate function which returns []*UserRecord. Let me know your thoughts. Thanks.

Comment (CDDelta) [reply auth/user_mgt.go]: Yes, I think you would have to do that. I am unsure how you would have to format the errors though. Maybe someone from the Firebase team can chime in here?

**Comment by hiranya911** [comment]
Thanks @vr001  for the PR. A couple of points to note:

1. Since this is a new API, it has to go through our internal review processes. This will take some time, so please sit tight.
2. Like @CDDelta point out, we should try to fetch all the necessary data from just one RPC call. The way it is implemented now is not going to scale.

**Comment by vr001** [comment]
Thanks @hiranya911 for your response. 
I agree with the one RPC call for fetching all the users. Looking forward to hear regarding the plan (or proposal) for the new API.

**Comment by hiranya911** [comment]
The Auth team has a plan to implement this feature. That will enable querying multiple user IDs, emails and phone numbers in a single request. Closing this.

**Pull request closed without merging**